src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php

```diff
@@ -114,15 +119,30 @@ public function searchAction(Request $request)
      *     message="You do not have permission to add this."
      * )
      *
-     * @return RedirectResponse
+     * @param Request $request
+     *
+     * @return Response
      */
-    public function createAction()
+    public function createAction(Request $request)
     {
-        $legacyLink = $this->getAdminLink('AdminSuppliers', [
-            'addsupplier' => 1,
-        ]);
+        try {
+            $supplierForm = $this->getFormBuilder()->getForm();
+            $supplierForm->handleRequest($request);
 
-        return $this->redirect($legacyLink);
+            $result = $this->getFormHandler()->handle($supplierForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_suppliers_index');
+            }
+        } catch (CoreException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Suppliers/add.html.twig', [
```

Comment:
@PierreRambaud Is it better to use `PrestaShopBundle:` instead of `@PrestaShop` here or does it change nothing ?

Reply:
Better to use `@PrestaShop` if you want to override only the template.